src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Sockets.Tests
+{
+    public class UnixDomainSocketTest : IDisposable
+    {
+        private readonly ITestOutputHelper _log;
+        private readonly string _path;
+        private readonly UnixDomainSocketEndPoint _endPoint;
+
+        public UnixDomainSocketTest(ITestOutputHelper output)
+        {
+            _log = TestLogging.GetInstance();
+            _path = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            if (File.Exists(_path))
+            {
+                File.Delete(_path);
+            }
+
+            _endPoint = new UnixDomainSocketEndPoint(_path);
+        }
+
+        private void OnConnectAsyncCompleted(object sender, SocketAsyncEventArgs args)
+        {
+            ManualResetEvent complete = (ManualResetEvent)args.UserToken;
+            complete.Set();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Socket_CreateUnixDomainSocket_Throws_OnWindows()
+        {
+            Assert.False(Capability.UnixDomainSocketsSupport());
+            Assert.Throws<SocketException>(() => new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified));
```

Comment:
> Technically, this test is not needed, but I wanted to document the behavior of Socket(AddressFamily.Unix,...) on Windows

This test is required as any behavior should be pinned by the tests.

> Given that there is no change in the behavior, and it's common to check the exception type and not the message in the code, the test only checks for the exception type.

I'm not asking for the code to be updated. I'm only asking for the exception message since there are cases where developers would import other libraries that depend on Sockets. 
If these libraries have taken a dependency on Unix sockets (or have been somehow incorrectly configured by the application) the message should be clear enough for the developer to correct the mistake (if possible) or understand that the library cannot be used on Windows.